src/main/webapp/scripts/group-details-script.js

```diff
@@ -0,0 +1,314 @@
+  * obj1: object you are comparing to default
+  * obj2: default object
+ */
+function diff(obj1, obj2) {
+    var diffs = {};
+    for (key in obj2) {
+        if (obj2.hasOwnProperty(key)) {
+            if (obj1[key] !== obj2[key]) {
+                diffs[key] = obj1[key];
+            }
+        }
+    }
+    return diffs;
+}
+
+/** Function called when user clicks to view access and membership settings */
+function viewSettings() {
+    var settingsChart = document.getElementById("group-settings-chart");
+    if (settingsChart.classList.contains("hidden")) {
+        settingsChart.classList.remove("hidden")
+    } else {
+        settingsChart.classList.add("hidden")
+    }
+}
+
+/** Fills in the checkbox table for access and membership settings */
+function setAccessMembershipSettingsTable(json) {
+    // create the access and membership settings table chart
+    var accessSettingsCheckboxes = [
+            [document.getElementById("contact-owners-group-owners"),
+            document.getElementById("contact-owners-group-managers"),
+            document.getElementById("contact-owners-group-members"),
+            document.getElementById("contact-owners-entire-organization"),
+            document.getElementById("contact-owners-external"),],
+            [document.getElementById("view-members-group-owners"),
+            document.getElementById("view-members-group-managers"),
+            document.getElementById("view-members-group-members"),
+            document.getElementById("view-members-entire-organization"),],
+            [document.getElementById("view-topics-group-owners"),
+            document.getElementById("view-topics-group-managers"),
+            document.getElementById("view-topics-group-members"),
+            document.getElementById("view-topics-entire-organization"),
+            document.getElementById("view-topics-external"),],
+            [document.getElementById("publish-posts-group-owners"),
+            document.getElementById("publish-posts-group-managers"),
+            document.getElementById("publish-posts-group-members"),
+            document.getElementById("publish-posts-entire-organization"),
+            document.getElementById("publish-posts-external"),],
+        ]
+    accessSettingsChecked = [
+            [true, false, false, false, false,],
+            [true, false, false, false,],
+            [true, false, false, false, false,],
+            [false, false, false, false, false,],
+        ]
+    var membershipSettingsCheckboxes = [
+            document.getElementById("manage-members-group-owners"),
+            document.getElementById("manage-members-group-managers"),
+            document.getElementById("manage-members-group-members"),
+        ]
+    membershipSettingsChecked = [
+            false, false, false,
+        ]
+
+    // contact owners
+    if (json.whoCanContactOwner == "ANYONE_CAN_CONTACT") {
```

Comment:
Large sets of if statements can usually be made more readable with a switch case https://www.w3schools.com/js/js_switch.asp block

Reply:
i'm considering getting rid of the if statements entirely